### PR TITLE
[CM T1] Fix errors in tutorials for Cascading Lookups and Command-line Options

### PR DESCRIPTION
**Pull request by flo91**
This T1-PR was done by me, Florian Lindner.

## Basics

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [x] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned [every code](/.reuse/dep5) not directly written by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

- [ ] Add the "work in progress" label if you do not want the PR to be reviewed yet.
- [ ] Add the "ready to merge" label **if the basics are fulfilled** and no further pushes are planned by you.

### CODE REVIEW

**Comment by flo91** [comment]
jenkins build libelektra please

**Review by kodebach — COMMENTED** [review]
Great work.

If you have time, you could look into the (markdown) shell recorder and why there is no automated test for the snippets in `doc/tutorials/cascading.md`.

File: doc/tutorials/cascading.md [hunk 1]
```diff
@@ -176,9 +176,8 @@ kdb get /tests/tutorial/cascading/#0/current/test
 As we used a cascading key for our override link (`/tests/overrides/test`) we can use this to allow users to provide their own `tests/overrides/test` keys. If a user sets the `/tests/overrides/test` key, the **user** namespace will be used (for a non-root user) and therefore the new target for our `/tests/tutorial/cascading/#0/current/test` key will be `user:/tests/overrides/test` instead of `system:/tests/overrides/test`.
 
 ```sh
-kdb set /tests/overrides/test "hello user"
-#> Set string to "hello user"
-#> Using name system:/tests/overrides/test
+kdb set user:/tests/overrides/test "hello user"
+#> Create a new key user:/tests/overrides/test with string "hello user"
```
[the remark this answers is not in the document]
Comment (flo91) [reply doc/tutorials/cascading.md:180]: Indeed, very interesting.
I tried to replace my change with the old content:
```
kdb set /tests/overrides/test "hello user"
#> Set string to "hello user"
```
and the test in Docker (with `make run_all`) failed, as it should be.
After changing it back the test was successful.
So it seems that the test is now working correctly.

[the remark this answers is not in the document]
Comment (flo91) [reply doc/tutorials/cascading.md:180]: Indeed, very interesting.
I tried to replace my change with the old content:
```
kdb set /tests/overrides/test "hello user"
#> Set string to "hello user"
```
and the test in Docker (with `make run_all`) failed, as it should be.
After changing it back the test was successful.
So it seems that the test is now working correctly.

**Comment by flo91** [comment]
> Great work.
> 
> If you have time, you could look into the (markdown) shell recorder and why there is no automated test for the snippets in `doc/tutorials/cascading.md`.

The test is defined in 
https://github.com/ElektraInitiative/libelektra/blob/f2b9d555a2de6da944781bd9edb4a2a877cd3c35/tests/shell/shell_recorder/tutorial_wrapper/CMakeLists.txt#L27

and also gets executed. (at least in Docker with `make run_all`)

**Comment by flo91** [comment]
> Great work.
> 
> If you have time, you could look into the (markdown) shell recorder and why there is no automated test for the snippets in `doc/tutorials/cascading.md`.

The test is defined in 
https://github.com/ElektraInitiative/libelektra/blob/f2b9d555a2de6da944781bd9edb4a2a877cd3c35/tests/shell/shell_recorder/tutorial_wrapper/CMakeLists.txt#L27

and also gets executed. (at least in Docker with `make run_all`)

**Pull request merged**